### ISSUE TITLE
Remove routing via attention mask

### ISSUE THREAD

**Issue by Eupham** [Eupham/KernelDev#106]
I want all references to attention masks to be purged. By this I mean, I do not want one calculated in model.py, I do not want model.py to pass so much as a attention_mask=None to the kernel. I do not want the kernel to use or accept mask as an argument at all for routing or as an input. Only metadata should be passed, accepted, or used for attention behavior.

This does not mean get rid of the mask which the kernel builds, it means do not route or accept an external mask.  Use the metadata to build the mask.  The attention behavior is to be determined by the metadata which builds an internal mask.  We need to make sure to avoid redundant calculations
